internal/pkg/cfa/function.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cfa implements cross-function analysis. The analyzer
+// defined in this package looks at every function in the transitive
+// dependencies of the program being analyzed and creates an abstraction
+// for each one that can be used to determine what the function does with
+// each of its arguments.
+
+package cfa
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"golang.org/x/tools/go/ssa"
+)
+
+// A Function is an abstraction for a Go function.
+// It can be queried about what it does with its arguments.
+type Function interface {
+	// Does this function's nth argument reach a sink?
+	Sinks(arg int) bool
+
+	// If this argument is tainted, which return values are tainted?
+	Taints(arg int) []int
```

Comment:
Mentioned offline / in another thread, but this interface should also support argument tainting.  The taint on, for instance, [writing to a buffer](https://golang.org/pkg/bytes/#Buffer.Write) is not on the return values.

Reply:
I agree that this should be handled eventually, but for a first stab at cross-function analysis I think that would be too much. Even when the current PR is merged, #57 will have a large diff, and it is already rather complex, so I think it would be preferable to wait until a future PR.